generators/app/index.js

```diff
@@ -36,182 +52,158 @@ module.exports = class extends Generator {
-                    "Static webserver",
-                    "SAP Launchpad service",
-                    "SAP NetWeaver"
-                ],
-                default: "Static webserver"
+                name: "framework",
+                message: "Which framework do you want to use?",
+                choices: Object.keys(fwkInfo),
+                default: Object.keys(fwkInfo)[0]
             },
             {
-                type: "list",
-                name: "ui5libs",
-                message: "Where should your UI5 libs be served from?",
-                choices: (props) => {
-                    return ["Content delivery network (SAPUI5)", "Local resources (SAPUI5)"];
+                when: (response) => {
+                    this._minFwkVersion = fwkInfo[response.framework].minVersion;
+                    return true;
+                },
+                type: "input", // HINT: we could also use the version info from OpenUI5/SAPUI5 to provide a selection!
+                name: "frameworkVersion",
+                message: "Which framework version do you want to use?",
+                default: async (answers) => {
+                    const minVersion = fwkInfo[answers.framework].minVersion;
+                    const npmPackage = fwkInfo[answers.framework].npmPackage;
+                    try {
+                        return (
+                            await packageJson(npmPackage, {
+                                version: "*" // use highest version, not latest!
+                            })
+                        ).version;
+                    } catch (ex) {
+                        chalk.red("Failed to lookup latest version for ${npmPackage}! Fallback to min version...");
+                        return minVersion[answers.framework];
+                    }
                 },
-                default: "Content delivery network (SAPUI5)"
+                validate: (v) => {
+                    return (
+                        (v && semver.valid(v) && semver.gte(v, this._minFwkVersion)) ||
+                        chalk.red(
+                            `Framework requires the min version ${this._minFwkVersion} due to the availability of the NPM packages!`
+                        )
+                    );
+                }
             },
             {
                 type: "checkbox",
                 name: "features",
                 message: "Do you want to add sample code?",
-                choices: (props) => {
+                choices: () => {
                     return [
                         "Add button to launchpad header",
                         "Add a launchpad footer with button",
                         "Add buttons to Me Area"
                     ];
                 },
-                default:[]
+                default: []
+            },
+            {
+                type: "input",
+                name: "author",
+                message: "Who is the author of the application?",
```

Comment:
Is "application" the right term for an FLP plugin?

Reply:
changed it to FLP Plugin